tekton-proposal.md

```diff
@@ -0,0 +1,73 @@
+# Knative Build and Tekton Pipelines Steering Committee proposal
+
+Jason Hall ([jasonhall@google.com](mailto:jasonhall@google.com))
+
+Feb 25 2019
+
+## Objective
+
+Propose a resolution to the ongoing confusion around the future of Knative Build with regard to the recently announced Tekton Pipelines change. This doc should be used as a forum for external comment and questions, to resolve any concerns before the Steering Committee decides whether to adopt the proposal.
+
+Please leave comments on this PR. The text of the proposal may change based on feedback, and the PR will retain the history of these edits.
+
+## Background
+
+Knative Build was conceived at the beginning of the Knative project alongside the other serverless-focused components (Serving and Eventing) to enable a "serverless development experience". Its goal was to enable users to specify their source to a Knative Service deployment and the steps to take to make it a container, and deploy that container, while interacting with a single custom resource.
+
+From the beginning, Build had bigger ambitions. Its [2018 roadmap](https://github.com/knative/build/blob/master/roadmap-2018.md) mentions features that aren't directly related to furthering serverless deployments (triggering, workflow, etc.). These aren't accelerators for serverless deployments, these are CI/CD features!
+
+This tension has existed in Build from the beginning: did it exist to accelerate the Knative serverless platform? Or is it supposed to be a full-featured CI/CD platform? How should Build prioritize features that don't have anything to do with the rest of Knative? These are not hypothetical questions, these are questions we've heard from users, operators and contributors.
```

Comment:
That's a great point, thanks.

If the Knative community wants to invest in an offline data processing system to compliment its current short-lived synchronous frontend offering, that's entirely reasonable. I don't think that thing should be called a "Build" however, and might need more features than what Build currently focuses on. It might grow out of Serving's current model instead, toward something like [AWS Step Functions](https://aws.amazon.com/step-functions). It might even warrant a whole new set of resources and separate working group in the future.

So far, I don't know whether there's enough desire to embark on that effort right now, or enough strong opinions about how it would work to tell how it should be built. In time, it could even be built on Tekton.

Reply:
> I don't think that thing should be called a "Build" however, and might need more features than what Build currently focuses on. It might grow out of Serving's current model instead, toward something like [AWS Step Functions](https://aws.amazon.com/step-functions). It might even warrant a whole new set of resources and separate working group in the future.

Can we call it Task and Pipeline? Asking for a friend 😉

Reply:
I see what you did there. 😄 

I think they should be named whatever is clearest once we know what they do and how they work. Since the whole endeavor is hypothetical at this point I suppose we can call them whatever we want! 🐪